src/sprites/setup.py

```diff
@@ -0,0 +1,230 @@
+        """
+        self._direction_exceptions[direction] = rect
+
+    def set_state_exception(
+            self, state: EntityState, rect: pygame.Rect
+    ) -> None:
+        """
+        Tell the class, in which state a hitbox should be used that differs
+        from the default one.
+        """
+        self._state_exceptions[state] = rect
+
+    def set_exception(
+            self, state: EntityState, direction: Direction, rect: pygame.Rect
+    ) -> None:
+        """
+        Tell the class, in which state a hitbox should be used that differs
+        from the default one.
+
+        Hitboxes that were stored through this method will be used before any
+        hitbox specific to the current Entity state or direction is considered.
+        """
+        self._exceptions[(state, direction)] = rect
+
+    def get_hitbox(
+            self, state: EntityState, direction: Direction
+    ) -> pygame.Rect:
+        state_exception = self._state_exceptions.get(state)
+        direction_exception = self._direction_exceptions.get(direction)
+        exception = self._exceptions.get((state, direction))
+        if not exception:
+            if not direction_exception:
+                if not state_exception:
+                    return self.default
+                return state_exception
+            return direction_exception
+        return exception
+
+    def scale_hitboxes(self, factor: float) -> None:
+        print(self._state_exceptions, self._direction_exceptions, self._exceptions, self.default)
+        for item in [self._state_exceptions.values(),
+                     self._direction_exceptions.values(),
+                     self._exceptions.values(),
+                     [self.default]]:
+            for hitbox in item:
+                hitbox.update(
+                    hitbox.left * factor,
+                    hitbox.top * factor,
+                    hitbox.width * factor,
+                    hitbox.height * factor
+                )
+
+
+class EntityAssets:
+    Rabbit: EntityAsset
+
+    @classmethod
+    def setup(cls):
+
+        rabbit_hitbox = _Hitbox(pygame.Rect(18, 26, 12, 6))
+        rabbit_hitbox.set_direction_exception(
+            Direction.LEFT, pygame.Rect(20, 26, 8, 6)
+        )
+        rabbit_hitbox.set_direction_exception(
+            Direction.RIGHT, pygame.Rect(20, 26, 8, 6)
+        )
+        rabbit_hitbox.set_exception(
+            EntityState.AXE, Direction.LEFT, pygame.Rect(24, 26, 8, 6)
+        )
+        rabbit_hitbox.set_exception(
+            EntityState.AXE, Direction.RIGHT, pygame.Rect(16, 26, 8, 6)
+        )
+
+        rabbit_asset = entity_importer(
+            path="images/characters/rabbit",
+            size=48,
+            directions=[Direction.DOWN, Direction.UP, Direction.LEFT],
+            hitbox=rabbit_hitbox
+        )
+
+        cls.Rabbit = EntityAsset(
+            rabbit_asset
+        )
+
+
+_StateHitboxTypes = (
```

Comment:
I did not find any usage for this...

Reply:
Oh, that seems to be an artifact from an unnecessary and very cumbersome attempt, will delete it right away...